resources/providers/gcplib/inventory/provider.go

```diff
@@ -189,10 +258,109 @@ func mergeAssetContentType(assets []*assetpb.Asset) []*assetpb.Asset {
 			item.IamPolicy = asset.IamPolicy
 		}
 	}
-
 	var results []*assetpb.Asset
 	for _, asset := range resultsMap {
 		results = append(results, asset)
 	}
 	return results
 }
+
+// extends the assets with the project and organization display name
+func extendAssets(ctx context.Context, crm *ResourceManagerWrapper, assets []*assetpb.Asset) []*ExtendedGcpAsset {
+	var extendedAssets []*ExtendedGcpAsset
+	var cache = make(map[string]*fetching.EcsGcpCloud)
+	for _, asset := range assets {
+		keys := getAssetIds(asset)
+		if ecsGcpCloudCached, ok := cache[keys.cacheKey]; ok {
+			extendedAssets = append(extendedAssets, &ExtendedGcpAsset{
+				Asset:       asset,
+				EcsGcpCloud: ecsGcpCloudCached,
+			})
+			continue
+		}
+		cache[keys.cacheKey] = getEcsGcpCloudData(ctx, crm, keys)
+		extendedAssets = append(extendedAssets, &ExtendedGcpAsset{
+			Asset:       asset,
+			EcsGcpCloud: cache[keys.cacheKey],
+		})
+	}
+	return extendedAssets
+}
+
+type GcpAssetIDs struct {
+	orgId         string
+	projectId     string
+	parentProject string
+	parentOrg     string
+	cacheKey      string
+}
+
+func getAssetIds(asset *assetpb.Asset) GcpAssetIDs {
+	orgId := getOrganizationId(asset.Ancestors)
+	projectId := getProjectId(asset.Ancestors)
+	parentProject := fmt.Sprintf("projects/%s", projectId)
+	parentOrg := fmt.Sprintf("organizations/%s", orgId)
+	return GcpAssetIDs{
+		orgId:         orgId,
+		projectId:     projectId,
+		parentProject: parentProject,
+		parentOrg:     parentOrg,
+		cacheKey:      fmt.Sprintf("%s/%s", parentProject, parentOrg),
```

Comment:
It makes more sense to me to generate the cache key out of this function response and not include it in the `GcpAssetIDs` struct.